### PR TITLE
feat: Disable location sharing for C1 - WPB-9749

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-9749" title="WPB-9749" target="_blank">BugWPB-9749</a>  [iOS] Location sharing should be disabled in C1 builds
  </td></table>
  <br />
 

### Issue
Disable location sharing for C1

**Needs releases with:**
https://github.com/wireapp/wire-ios-build-assets/pull/114

### Testing
If `LOCATION_SHARING_ENABLED` is false, the user shouldn't see the location button.

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

### CODE REVIEW

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift [hunk 1]
```diff
@@ -252,9 +253,20 @@ final class ConversationInputBarViewController: UIViewController,
...
+        if shouldExcludeLocationButton {
+            if let index = buttonsArray.firstIndex(of: locationButton) {
+                buttonsArray.remove(at: index)
+            }
+        }
+
         return buttonsArray
     }
 
+    private var shouldExcludeLocationButton: Bool {
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift:266]:
```suggestion
    /// Remove locationButton if security flag does not allow it
    private var shouldExcludeLocationButton: Bool {
```

File: wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift [hunk 2]
```diff
@@ -252,9 +253,20 @@ final class ConversationInputBarViewController: UIViewController,
             buttonsArray.insert(hourglassButton, at: buttonsArray.startIndex)
         }
 
+        // Remove locationButton if security flag does not allow it
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController/ConversationInputBarViewController.swift]:
```suggestion
```

**Review by netbe — APPROVED** [review]
LGTM;) don't forget to update wire-ios-build-assets too? added a small suggestion

**Review by markusfassbender — APPROVED** [review]

**Review by johnxnguyen — APPROVED** [review]

**Comment by KaterinaWire** [comment]
> LGTM;) don't forget to update wire-ios-build-assets too? added a small suggestion

I updated `wire-ios-build-assets` version to 42f724c (the last commit with merged PR)

**Review by netbe — APPROVED** [review]

**Pull request merged**